src/window-service.js

```diff
@@ -195,7 +147,13 @@
 
             var mainWindow;
             if (name) {
-                mainWindow = new fin.desktop.Window(createConfig(name, isCompact), () => {
+                var config;
+                if (isCompact) {
+                    config = this.configService.getCompactConfig(name);
+                } else {
+                    config = this.configService.getWindowConfig(name);
+                }
```

Comment:
I decided against that because the line would get quite long and potentially confusing. I don't mind, though